src/app/zap-templates/templates/app/cluster-objects-src.zapt

```diff
@@ -80,6 +104,24 @@ CHIP_ERROR TypeInfo::DecodableType::Decode(TLV::TLVReader &reader, const Concret
     {
 {{#zcl_attributes_server}}
     case Attributes::{{asUpperCamelCase label}}::TypeInfo::GetAttributeId():
+        {{#if_is_enum type}}
+        {{#unless isArray}}
+        {{#if (isWeaklyTypedEnum (asUpperCamelCase type))}}
+// Need to convert consumers to using the new enum classes, so we
+// don't just have casts all over.
+#ifdef CHIP_USE_ENUM_CLASS_FOR_IM_ENUM
+        {{/if}}
+        {{#if isNullable}}
+        VerifyOrReturnError(!{{asLowerCamelCase label}}.IsNull(), CHIP_NO_ERROR);
+        ReturnErrorOnFailure(Check{{asUpperCamelCase type}}({{asLowerCamelCase label}}.Value()));
```

Comment:
Again, none of this makes sense.  I ask again: was this tested?

Reply:
updated, thanks